bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVControl.java

```diff
@@ -0,0 +1,582 @@
+                thingConfig.getVerificationCode());
+        String soapXml = MessageFormat.format(SOAP_ENVELOPE, soapBody);
+        String response = http.httpPOST(buildHost(), buildReceiverUrl(PAIRCHECK_URI), soapXml, PAIRCHECK_SOAP_ACTION,
+                CONNECTION_CLOSE);
+
+        // Exception if request failed (response code != HTTP_OK)
+        if (!response.contains("<pairingResult>")) {
+            throw new MagentaTVException("Unexpected result for pairing verification: " + response);
+        }
+
+        String result = getXmlValue(response, "pairingResult");
+        if (!result.equals("0")) {
+            logger.info("Pairing failed or pairing no longer valid, result={0}", result);
+            resetPairing();
+            // let the caller decide how to proceed
+            return false;
+        }
+
+        if (!thingConfig.isMR400()) {
+            String enable4K = getXmlValue(response, "Enable4K");
+            String enableSAT = getXmlValue(response, "EnableSAT");
+            logger.info("Features for device {0}: Enable4K:{1}, EnableSAT:{2}", thingConfig.getFriendlyName(), enable4K,
+                    enableSAT);
+        }
+        return true;
+    }
+
+    /**
+     *
+     * @return true if pairing is completed (verification code was generated)
+     */
+    public boolean isPaired() {
+        // pairing was completed successful if we have the verification code
+        return !thingConfig.getVerificationCode().isEmpty();
+    }
+
+    /**
+     * Reset pairing information (e.g. when verification failed)
+     */
+    public void resetPairing() {
+        // pairing no longer valid
+        thingConfig.setPairingCode("");
+        thingConfig.setVerificationCode("");
+    }
+
+    /**
+     * Send key code to the MR (via SOAP request). A key code could be send by it's
+     * code (0x.... notation) or with a symbolic namne, which will first be mapped
+     * to the key code
+     *
+     * XML format for Send Key
+     *
+     * <s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\"
+     * s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\"> <s:Body>\n
+     * <u:X_CTC_RemoteKey
+     * xmlns:u=\"urn:schemas-upnp-org:service:X-CTC_RemoteControl:1\">\n
+     * <InstanceID>0</InstanceID>\n
+     * <KeyCode>keyCode=$keyCode^$pairingDeviceID:$verificationCode^userID:$userID</KeyCode>\n
+     * </u:X_CTC_RemoteKey>\n </s:Body></s:Envelope>
+     *
+     * @param keyName
+     * @return true: successful, false: failed, e.g. unkown key code
+     * @throws MagentaTVException
+     */
+    public boolean sendKey(String keyName) throws MagentaTVException {
+
+        String keyCode = getKeyCode(keyName);
+        logger.debug("Send Key {0} (keyCode={1}) to device {2} ({3})", keyName, keyCode, thingConfig.getFriendlyName(),
+                thingConfig.getTerminalID());
+        if (keyCode.length() <= "0x".length()) {
+            logger.info("Key {0} is unkown!", keyCode);
```

Comment:
change to logger .debug

Reply:
first one changed, 2nd is important info for the user

Reply:
How could one send an unknown key code?